src/core/client.cpp

```diff
@@ -26,52 +26,41 @@
 #include "message.hpp"
 #include "state.hpp"
 #include "use.hpp"
-#include "websockets.hpp"
 #include <deepstream/core/buffer.hpp>
 #include <deepstream/core/client.hpp>
-#include <deepstream/core/error_handler.hpp>
 
 #include <cassert>
-#include <include/deepstream.hpp>
 
 namespace deepstream {
 
-Client::Client(const std::string& uri, std::unique_ptr<ErrorHandler> p_eh, WSFactory* wsFactory)
-    : p_impl_(Connection::make(uri, std::move(p_eh), nullptr).release())
+Client::Client(const std::string &uri, WSHandler &ws_handler, ErrorHandler &error_handler)
+    : p_connection_(new Connection(uri, ws_handler, error_handler, event, presence))
     , event([this](const Message& message) -> bool {
-        assert(p_impl_);
-        return p_impl_->send_(message) == websockets::State::OPEN;
+        assert(p_connection_);
+        p_connection_->send(message);
```

Comment:
send is injected into the event handler to make unit testing simpler. it shouldn't really return a bool but didn't get round to changing it. (hence always returns true)